src/status_im/multiaccounts/recover/core.cljs

```diff
@@ -23,9 +23,9 @@
   (contains? multiaccounts (:key-uid root-key)))
 
 (defn check-phrase-warnings [recovery-phrase]
+  (log/debug (types/json->clj (status/validate-mnemonic recovery-phrase)))
   (cond (string/blank? recovery-phrase) :required-field
-        (not (mnemonic/valid-words? recovery-phrase)) :recovery-phrase-invalid
-        (not (mnemonic/status-generated-phrase? recovery-phrase)) :recovery-phrase-unknown-words))
+        (when-not (= (get-in (types/json->clj (status/validate-mnemonic recovery-phrase)) [:error]) "") true) :recovery-phrase-invalid))
```

Comment:
you shouldn't use synchronous methods and make validate-mnemonic an effect with a callback

Reply:
Is there a series concern about performance or is there just a desire to have all status-go calls be asynch? Feels like adding an asynch layer to this flow adds a lot of overhead.

Reply:
@yenda Implemented an async version of validate-mnemonic and it seems to be working well.  Let me know if there's anything you'd change in my approach.  Thanks!